lib/restaurant.rb

```diff
@@ -0,0 +1,56 @@
+class Restaurant 
+  attr_reader :order_history, :order, :text
+  attr_accessor :calc
+
+  def initialize(order_class = Order, text_class = Text, sender_class = Sender)
+    @order_class = order_class
+    @order_history = []
+    @order = @order_class.new
+    @calc = 0
+    @text_class = text_class
+    @text = nil
+    @sender_class = sender_class
+    @sender = @sender_class.new
+  end 
+
+  def view_menu 
+    @order.menu_listing
+  end
+
+  def start_order
+    @order = @order_class.new
+  end
+
+  def scrap_order
+    @order = @order_class.new
+  end
+
```

Comment:
@order = nil might be more useful, for **def initialize** and **def scrap_order**